libs/core/src/product/containers/product/product.component.ts

```diff
@@ -35,4 +36,8 @@ export class ProductContainer implements OnInit {
       select(fromProduct.selectSelectedProduct)
     );
   }
+
+  addToCart(payload) {
```

Comment:
This isn't the responsibility of "product". I'd argue you'd probably want to keep all "cart" related things in cart. It doesn't make sense to breach the boundaries of these two distinctly separate modules.

Reply:
Been thinking about how else to do this, and I'm not sure how exactly to do it if it isn't here. The problem I'm running into is that I can only have one attribute component per `div`. So I can't just add it to the daffodil/core cart-container, and do something like this:

```
<div class="product-container"
  product-container #ProductContainer="ProductContainer" [selectedProductId]="productId"
  cart-container #CartContainer="CartContainer">
    <product [product]="ProductContainer.product$ | async" (addToCart)="CartContainer.addToCart($event)" *ngIf="!(ProductContainer.loading$ | async)"></product>
    <div *ngIf="(ProductContainer.loading$ | async)" class="product-container__loading-icon">
      <i class="fa fa-spinner fa-spin fa-5x"></i>
    </div>
</div>
```

This could work:
```
<div class="product-container" product-container #ProductContainer="ProductContainer" [selectedProductId]="productId">
  <div class="cart-container" cart-container #CartContainer="CartContainer">
    <product [product]="ProductContainer.product$ | async" (addToCart)="CartContainer.addToCart($event)" *ngIf="!(ProductContainer.loading$ | async)"></product>
    <div *ngIf="(ProductContainer.loading$ | async)" class="product-container__loading-icon">
      <i class="fa fa-spinner fa-spin fa-5x"></i>
    </div>
  </div>
</div>
```
but yikes.

I could just have a service in daffodil/core that dispatches the action, but I'd still need to import that service into the product-container, which is basically the same thing with an extra step. This definitely seems like something you should be able to do in Angular, but it seems that everything I think to try, I eventually get to a roadblock that basically says, "that's not how angular works". Any ideas?

Reply:
So, I think this may be a case for some nifty transclusion. The `product` component doesn't need to be aware of "AddToCart". That's not its responsibility. So, I'd migrate the component out, yet allow it to transclude something that has the `add-to-cart` attribute on it.

Reply:
done. I only did the bare minimum here. I'll need to add some other stuff if I ever need to transclude anything else, but I figure only add what I need